packages/next/client/image.tsx

```diff
@@ -428,6 +429,16 @@ function akamaiLoader({ root, src, width }: LoaderProps): string {
   return `${root}${normalizeSrc(src)}?imwidth=${width}`
 }
 
+function contentfulLoader({ src, quality, width }: LoaderProps): string {
+  const params = ['w=' + width]
```

Comment:
What about auto formatting (webp) and max width limiting? (see cloudinary/imgix below)

Reply:
Unfortunately, these are params the [Contentful Images API](https://www.contentful.com/developers/docs/references/images-api/#/reference/deletion) doesn't offer.

Reply:
This could lead to upscaling source images and actually increase the size instead of the expected decrease.

What happens when you have a source image that's 500 pixels wide but you pass `w=1000`?

Reply:
That was a great question. 👍 The Contentful Images API is not upscaling and keeps the max uploaded image dimensions even when `w` is greater than the dimension.

Example:

- [original image](https://images.ctfassets.net/f20lfrunubsq/6E6lO4YbOiZJsD48z5ks6/8eb86af58492e4915fe816a478d20a20/Screenshot_2020-08-30_at_18.43.36.png) – 420k in size with 1378 × 832
- [image with `w=2000`](https://images.ctfassets.net/f20lfrunubsq/6E6lO4YbOiZJsD48z5ks6/8eb86af58492e4915fe816a478d20a20/Screenshot_2020-08-30_at_18.43.36.png?w=2000) - also 420k with 1378 × 832

Reply:
> Unfortunately, these are params the [Contentful Images API](https://www.contentful.com/developers/docs/references/images-api/#/reference/deletion) doesn't offer.

@stefanjudis I assume that means there is no way of leveraging Contentful webp formatting parameter? It's a shame that Contentful don't have an auto quality like most CDN's these days as it plays a significant part on size for users with supported browsers.

Reply:
@styfle be sure to pass a quality parameter `q` when you are delivering images via Contentful's Images API; the quality is set to 100% without it, resulting in much larger images. This is true of WebP and JPG. I recommend experimenting with `q` values from 30-70.